dbt/include/trino/macros/materializations/incremental.sql

```diff
@@ -1,21 +1,74 @@
-{% macro dbt_trino_get_append_sql(tmp_relation, target_relation) %}
+{% macro validate_get_incremental_strategy(config) %}
+  {#-- Find and validate the incremental strategy #}
+  {%- set strategy = config.get("incremental_strategy", default="append") -%}
 
-    {%- set dest_columns = adapter.get_columns_in_relation(target_relation) -%}
+  {% set invalid_strategy_msg -%}
+    Invalid incremental strategy provided: {{ strategy }}
+    Expected one of: 'append', 'delete+insert'
+  {%- endset %}
+  {% if strategy not in ['append', 'delete+insert'] %}
+    {% do exceptions.raise_compiler_error(invalid_strategy_msg) %}
+  {% endif %}
+
+  {% do return(strategy) %}
+{% endmacro %}
+
+{% macro get_incremental_sql(strategy, tmp_relation, target_relation, unique_key, dest_columns) %}
+  {% if strategy == 'append' %}
+    {% do return(get_append_sql(target_relation, tmp_relation, dest_columns)) %}
+  {% elif strategy == 'delete+insert' %}
+    {% do return(get_delete_insert_merge_sql(target_relation, tmp_relation, unique_key, dest_columns)) %}
+  {% endif %}
+{% endmacro %}
+
+{% macro get_append_sql(target_relation, tmp_relation, dest_columns) %}
     {%- set dest_cols_csv = dest_columns | map(attribute='quoted') | join(', ') -%}
     insert into {{ target_relation }}
-    select {{dest_cols_csv}} from {{ tmp_relation.include(database=true, schema=true) }};
+    select {{dest_cols_csv}} from {{ tmp_relation }};
 
     drop table if exists {{ tmp_relation }};
-
 {% endmacro %}
 
-{% materialization incremental, adapter='trino' -%}
+{% macro trino__get_delete_insert_merge_sql(target, source, unique_key, dest_columns) -%}
+    {%- set dest_cols_csv = get_quoted_csv(dest_columns | map(attribute="name")) -%}
+
+    {% if unique_key %}
+        {% if unique_key is sequence and unique_key is not string %}
+            delete from {{ target }}
+            where
+                {% for key in unique_key %}
+                    {{ target }}.{{ key }} IN (SELECT {{ key }} FROM {{ source }})
```

Comment:
most of the statements here use lowercase for the reserved sql keywords.
pls change `IN` `SELECT and `FROM` accordingly.